NDTensors/src/lib/TypeParameterAccessors/src/base/ndims.jl

```diff
@@ -0,0 +1,17 @@
+## NDTensors.ndims (not imported from Base)
+
+## TODO So here I try to use the new type_parameters system for `NDTensors.ndims`
+## But if `ndims` is not defined for a type, I revert to using Base.ndims
+function TypeParameterAccessors.ndims(array)
```

Comment:
I am not sure this is the best way to implement this function. I haven't checked lower julia version so this might break on julia 1.6

Reply:
No matter how we implement it, let's definitely not use `try ... catch ...` for this (I very rarely come across a good use for that, that can't be implemented in a better way). What about just `type_parameter(array, ndims)`? Also, it seems better to me to base the definition of `TypeParameterAccessors.ndims` off of accessing the type parameter position name `TypeParameterAccessors.ndims`.

Additionally, we should make sure that either `TypeParameterAccessors.ndims` or `Base.ndims` can be used interchangeably as type parameter position names in any appropriate `TypeParameterAccessors` function.

Reply:
See a related comment https://github.com/ITensor/ITensors.jl/pull/1353/files#r1527215937.

Reply:
As discussed above, we removed the custom `ndims` functions and rely on `TypeParameterAccessors.type_paramter(type, ndims)`